presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergMetadata.java

```diff
@@ -662,4 +697,332 @@ public TableStatistics getTableStatistics(ConnectorSession session, ConnectorTab
+            getDataPath(icebergTable.location()),
+            getFileFormat(icebergTable));
+    }
+
+    @Override
+    public Optional<ConnectorOutputMetadata> finishRefreshMaterializedView(ConnectorSession session,
+            ConnectorInsertTableHandle insertHandle,
+            Collection<Slice> fragments,
+            Collection<ComputedStatistics> computedStatistics,
+            List<ConnectorTableHandle> sourceTableHandles)
+    {
+        IcebergWritableTableHandle table = (IcebergWritableTableHandle) insertHandle;
+        org.apache.iceberg.Table icebergTable = transaction.table();
+        List<CommitTaskData> commitTasks = new ArrayList<>();
+
+        if (!fragments.isEmpty()) {
+            commitTasks = fragments.stream()
+                .map(slice -> commitTaskCodec.fromJson(slice.getBytes()))
+                .collect(toImmutableList());
+
+            Type[] partitionColumnTypes = icebergTable.spec().fields().stream()
+                .map(field -> field.transform().getResultType(
+                    icebergTable.schema().findType(field.sourceId())))
+                .toArray(Type[]::new);
+
+            AppendFiles appendFiles = transaction.newFastAppend();
+            for (CommitTaskData task : commitTasks) {
+                DataFiles.Builder builder = DataFiles.builder(icebergTable.spec())
+                        .withInputFile(HadoopInputFile.fromLocation(task.getPath(), getConfiguration(session, table.getSchemaName())))
+                        .withFormat(table.getFileFormat())
+                        .withMetrics(task.getMetrics().metrics());
+
+                if (!icebergTable.spec().fields().isEmpty()) {
+                    String partitionDataJson = task.getPartitionDataJson()
+                            .orElseThrow(() -> new VerifyException("No partition data for partitioned table"));
+                    builder.withPartition(PartitionData.fromJson(partitionDataJson, partitionColumnTypes));
+                }
+
+                appendFiles.appendFile(builder.build());
+            }
+            appendFiles.commit();
+        }
+
+        // Update the map that tracks tables on which the materialized view depends and the corresponding snapshot ids of the table
+        UpdateProperties updateProperties = transaction.updateProperties();
+        Map<String, Long> tableSnapshotIdMap = new HashMap<>();
+        for (ConnectorTableHandle handle : sourceTableHandles) {
+            Optional<io.prestosql.plugin.iceberg.TableToken> token = getTableToken(session, handle);
+            if (token.isPresent()) {
+                tableSnapshotIdMap.put(handle.toString(), token.get().getSnapshotId());
+            }
+            else {
+                tableSnapshotIdMap.put(handle.toString(), (long) -1);
+            }
+        }
+        String tableSnapshotIDString = Joiner.on(",").withKeyValueSeparator("=").join(tableSnapshotIdMap);
+        updateProperties.set(DEPENDS_ON_TABLES, tableSnapshotIDString);
```

Comment:
This code stores the `dependsOnTables` metadata in the table properties of the Iceberg storage table. This solution works since the data insert and updating the table property is done in a transaction. 
An alternative way is to store the `dependsOnTables` metadata as snapshot metadata. The snapshot data and metadata are written at the same time. An additional advantage is that this leaves enough bread-crumbs such that if the materialized view is rolled back to a particular snapshot,  snapshot metadata has information on which snapshots of source tables were used to compute that snapshot of MV. This can be useful for debugging purposes. 
I will make this change as a separate commit on this PR in order to not disrupt ongoing review.